airflow/utils/context.py

```diff
@@ -161,54 +176,57 @@ class OutletEventAccessor:
     :meta private:
     """
 
-    raw_key: str | Asset | AssetAlias
+    key: BaseAssetUniqueKey
     extra: dict[str, Any] = attrs.Factory(dict)
     asset_alias_events: list[AssetAliasEvent] = attrs.field(factory=list)
 
-    def add(self, asset: Asset | str, extra: dict[str, Any] | None = None) -> None:
+    def add(self, asset: Asset, extra: dict[str, Any] | None = None) -> None:
         """Add an AssetEvent to an existing Asset."""
-        if isinstance(asset, str):
-            asset_uri = asset
-        elif isinstance(asset, Asset):
-            asset_uri = asset.uri
-        else:
+        if not isinstance(self.key, AssetAliasUniqueKey):
             return
 
-        if isinstance(self.raw_key, str):
-            asset_alias_name = self.raw_key
-        elif isinstance(self.raw_key, AssetAlias):
-            asset_alias_name = self.raw_key.name
-        else:
-            return
-
-        event = AssetAliasEvent(asset_alias_name, asset_uri, extra=extra or {})
+        asset_alias_name = self.key.name
+        event = AssetAliasEvent(
+            source_alias_name=asset_alias_name,
+            dest_asset_key=AssetUniqueKey.from_asset(asset),
+            extra=extra or {},
+        )
         self.asset_alias_events.append(event)
 
 
-class OutletEventAccessors(Mapping[str, OutletEventAccessor]):
+class OutletEventAccessors(Mapping[Union[Asset, AssetAlias], OutletEventAccessor]):
```

Comment:
I wonder if we should support int here… Not important either way, we can do it later.

Reply:
we probably should. but I think it might be better if we support it in the next PR. the purpose of this PR is adding `Asset.name` support to existing feature

Reply:
Sounds good. We should also support AssetRef.

Reply:
Yep, track it here https://github.com/apache/airflow/issues/44834